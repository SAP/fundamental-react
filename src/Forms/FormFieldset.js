import classnamesBind from 'classnames/bind';
import PropTypes from 'prop-types';
import React from 'react';
import withStyles from '../utils/withStyles';
import styles from 'fundamental-styles/dist/fieldset.css';

const classnames = classnamesBind.bind(styles);
/** A **FormFieldset** is used to give semantic meaning to a group of elements inside a form (e.g. Billing or Shipping Address).
 * Grouping fields together into a fieldset also provides styling and accessibility benefits. */
const FormFieldset = React.forwardRef(({ children, className, cssNamespace, ...props }, ref) => {

    const formFieldsetClasses = classnames(
        `${cssNamespace}-fieldset`,
        className
    );

    return (
        <fieldset
            {...props}
            className={formFieldsetClasses}
            ref={ref}>
            {children}
        </fieldset>
    );
});

FormFieldset.displayName = 'FormFieldset';

FormFieldset.propTypes = {
    /** Node(s) to render within the component */
    children: PropTypes.node,
    /** CSS class(es) to add to the element */
    className: PropTypes.string
};

export default withStyles(FormFieldset);
