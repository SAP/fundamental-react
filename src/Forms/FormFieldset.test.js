import FormFieldset from './FormFieldset';
import { mount } from 'enzyme';
import React from 'react';

describe('<FormFieldset />', () => {
    describe('FormFieldset Prop spreading', () => {
        test('should allow props to be spread to the FormFieldset component', () => {
            const element = mount(<FormFieldset data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });

    test('forwards the ref', () => {
        let ref;
        class Test extends React.Component {
            constructor(props) {
                super(props);
                ref = React.createRef();
            }
            render = () => <FormFieldset ref={ref} />;
        }
        mount(<Test />);
        expect(ref.current.tagName).toEqual('FIELDSET');
        expect(ref.current.className).toEqual('fd-fieldset');
    });
});
