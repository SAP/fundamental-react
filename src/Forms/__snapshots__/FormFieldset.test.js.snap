// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormFieldset /> create form field set 1`] = `
<fieldset
  className="fd-fieldset"
>
  <legend
    className="fd-fieldset__legend blue"
  >
    Radio buttons
  </legend>
  <div
    className="blue"
  >
    <div
      className="fd-form-item"
      onChange={[Function]}
    >
      <label
        className="fd-form-label"
      >
        <input
          className="fd-radio"
          id="radio-1"
          name="radio-group-1"
          type="radio"
          value="radio-1"
        />
        Option 1
        
      </label>
    </div>
    <div
      className="fd-form-item"
      onChange={[Function]}
    >
      <label
        className="fd-form-label"
      >
        <input
          checked={true}
          className="fd-radio"
          id="radio-2"
          name="radio-group-1"
          type="radio"
          value="radio-2"
        />
        Option 2
        
      </label>
    </div>
    <div
      className="fd-form-item"
      onChange={[Function]}
    >
      <label
        className="fd-form-label"
      >
        <input
          className="fd-radio"
          id="radio-3"
          name="radio-group-1"
          type="radio"
          value="radio-3"
        />
        Option 3
        
      </label>
    </div>
  </div>
  <legend
    className="fd-fieldset__legend"
    legendText="Radio buttons disabled"
  />
  <div
    className="blue"
  >
    <div
      className="fd-form-item fd-form-item--inline"
      onChange={[Function]}
    >
      <label
        className="fd-form-label is-disabled"
      >
        <input
          checked={true}
          className="fd-radio"
          disabled={true}
          id="radio-4"
          name="radio-group-2"
          type="radio"
          value="radio-4"
        />
        Option 1
        
      </label>
    </div>
    <div
      className="fd-form-item fd-form-item--inline"
      onChange={[Function]}
    >
      <label
        className="fd-form-label is-disabled"
      >
        <input
          className="fd-radio"
          disabled={true}
          id="radio-5"
          name="radio-group-2"
          type="radio"
          value="radio-5"
        />
        Option 2
        
      </label>
    </div>
    <div
      className="fd-form-item fd-form-item--inline"
      onChange={[Function]}
    >
      <label
        className="fd-form-label is-disabled"
      >
        <input
          className="fd-radio"
          disabled={true}
          id="radio-6"
          name="radio-group-2"
          type="radio"
          value="radio-6"
        />
        Option 3
        
      </label>
    </div>
  </div>
  <legend
    className="fd-fieldset__legend"
    legendText="Inline Radio buttons"
  />
  <div
    className="blue"
  >
    <div
      className="fd-form-item fd-form-item--inline"
      onChange={[Function]}
    >
      <label
        className="fd-form-label"
      >
        <input
          className="fd-radio"
          id="radio-7"
          name="radio-group-3"
          type="radio"
          value="radio-7"
        />
        Option 1
        
      </label>
    </div>
    <div
      className="fd-form-item fd-form-item--inline"
      onChange={[Function]}
    >
      <label
        className="fd-form-label"
      >
        <input
          className="fd-radio"
          id="radio-8"
          name="radio-group-3"
          type="radio"
          value="radio-8"
        />
        Option 2
        
      </label>
    </div>
    <div
      className="fd-form-item fd-form-item--inline"
      onChange={[Function]}
    >
      <label
        className="fd-form-label"
      >
        <input
          checked={true}
          className="fd-radio"
          id="radio-9"
          name="radio-group-3"
          type="radio"
          value="radio-9"
        />
        Option 3
        
      </label>
    </div>
  </div>
  <div
    className="blue"
  >
    <div
      className="fd-form-item fd-form-item--inline"
      onChange={[Function]}
    >
      <label
        className="fd-form-label is-disabled"
      >
        <input
          className="fd-radio"
          disabled={true}
          id="radio-7"
          name="radio-group-4"
          type="radio"
          value="radio-7"
        />
        Option 1
        
      </label>
    </div>
    <div
      className="fd-form-item fd-form-item--inline"
      onChange={[Function]}
    >
      <label
        className="fd-form-label is-disabled"
      >
        <input
          className="fd-radio"
          disabled={true}
          id="radio-8"
          name="radio-group-4"
          type="radio"
          value="radio-8"
        />
        Option 2
        
      </label>
    </div>
    <div
      className="fd-form-item fd-form-item--inline"
      onChange={[Function]}
    >
      <label
        className="fd-form-label is-disabled"
      >
        <input
          checked={true}
          className="fd-radio"
          disabled={true}
          id="radio-9"
          name="radio-group-4"
          type="radio"
          value="radio-9"
        />
        Option 3
        
      </label>
    </div>
  </div>
  <legend
    className="fd-fieldset__legend"
    legendText="Checkboxes"
  />
  <div
    className="fd-form-item blue"
  >
    <input
      className="fd-input blue"
      id="checkbox-1"
      name="checkbox-name-1"
      type="checkbox"
      value=""
    />
    <label
      className="fd-form-label blue"
      forAttr="checkbox-1"
    >
      Option One
      
    </label>
  </div>
</fieldset>
`;

exports[`<FormFieldset /> create form field set 2`] = `
<fieldset
  className="fd-fieldset blue"
>
  <legend
    className="fd-fieldset__legend blue"
  >
    Radio buttons
  </legend>
  <div
    className="blue"
  >
    <div
      className="fd-form-item"
      onChange={[Function]}
    >
      <label
        className="fd-form-label"
      >
        <input
          className="fd-radio"
          id="radio-1"
          name="radio-group-5"
          type="radio"
          value="radio-1"
        />
        Option 1
        
      </label>
    </div>
    <div
      className="fd-form-item"
      onChange={[Function]}
    >
      <label
        className="fd-form-label"
      >
        <input
          checked={true}
          className="fd-radio"
          id="radio-2"
          name="radio-group-5"
          type="radio"
          value="radio-2"
        />
        Option 2
        
      </label>
    </div>
    <div
      className="fd-form-item"
      onChange={[Function]}
    >
      <label
        className="fd-form-label"
      >
        <input
          className="fd-radio"
          id="radio-3"
          name="radio-group-5"
          type="radio"
          value="radio-3"
        />
        Option 3
        
      </label>
    </div>
  </div>
</fieldset>
`;
